ui/app/AppLayouts/Wallet/controls/NetworkFilter.qml

```diff
@@ -37,16 +62,21 @@ StatusComboBox {
         // Persist selection between selectPopupLoader reloads
         property var currentModel: layer1Networks
         property int currentIndex: 0
-    }
 
-    Component.onCompleted: {
-        if (!multiSelection && d.currentModel.count > 0) {
-            d.selectedChainName = d.currentModel.rowData(d.currentIndex, "chainName")
-            d.selectedIconUrl = d.currentModel.rowData(d.currentIndex, "iconUrl")
+        function setChainProperties(networkModel, index) {
+            // Update selected chain properties:
+            d.currentModel = networkModel
+            d.currentIndex = index
+            d.selectedChainName = ModelUtils.get(d.currentModel, d.currentIndex, "chainName")
+            d.selectedIconUrl = ModelUtils.get(d.currentModel, d.currentIndex, "iconUrl")
```

Comment:
WDYT of making these properties readonly and use only bindings? Should keep us safe from inconsistency issues.

Reply:
Good idea!